src/lsm/grid.zig

```diff
@@ -536,17 +542,18 @@ pub fn GridType(comptime Storage: type) type {
                 grid.read_resolving = false;
             }
 
+            // Remove the "root" read so that the address is no longer actively reading / locked.
+            grid.read_queue.remove(read);
```

Comment:
@kprotty I'm unsure of this change. My reasoning was that if we need to allow issuing writes to this address in the root read, then we need to allow it in the resolves too, since there's no guarantee on which callback ends up being the root.

Reply:
The "root" terminology really referred to "the first read in `read_queue` with that address that will resolve the block". Although it's a good point that those queued to the "root" read could issue write_block()'s to the same address, so the root should be removed from read_queue before it starts processing callbacks.